File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/AttendanceColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AttendanceRow } from "./AttendanceRow";
import { AttendanceType } from "./AttendanceType";

export interface AttendanceColumns {
    Id: Column<AttendanceRow>;
    UserId: Column<AttendanceRow>;
    NameUsername: Column<AttendanceRow>;
    DateNTime: Column<AttendanceRow>;
    Type: Column<AttendanceRow>;
    Coordinates: Column<AttendanceRow>;
    PunchOutCoordinates: Column<AttendanceRow>;
    Location: Column<AttendanceRow>;
    ApprovedByUsername: Column<AttendanceRow>;
    PunchIn: Column<AttendanceRow>;
    PunchOut: Column<AttendanceRow>;
    Distance: Column<AttendanceRow>;
}

export class AttendanceColumns extends ColumnsBase<AttendanceRow> {
    static readonly columnsKey = 'Operations.Attendance';
    static readonly Fields = fieldsProxy<AttendanceColumns>();
}

[AttendanceType]; // referenced types

File: HRMS/HRMS.Web/Modules/Operations/Attendance/AttendanceColumns.cs
namespace HRMS.Operations.Columns;

[ColumnsScript("Operations.Attendance")]
[BasedOnRow(typeof(AttendanceRow), CheckNames = true)]
public class AttendanceColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string UserId { get; set; }
    public string NameUsername { get; set; }
    public DateTime DateNTime { get; set; }
    public int Type { get; set; }
    public string Coordinates { get; set; }
    public string PunchOutCoordinates { get; set; }
    public string Location { get; set; }
    public string ApprovedByUsername { get; set; }
    public DateTime PunchIn { get; set; }
    public DateTime PunchOut { get; set; }
    public double Distance { get; set; }
}